lazyflow/operators/opResize.py

```diff
@@ -0,0 +1,253 @@
+
+
+def ceil_roi(roi: NDArray) -> NDArray:
+    """
+    Expand roi to nearest integer
+    """
+    return np.array([np.floor(roi[0]), np.ceil(roi[1])])
+
+
+class OpResize(Operator):
+    """
+    Reimplements scikit-image.transform.resize as a lazyflow operator,
+    handling in particular the halo for antialiasing in the context of subregions.
+
+    The gaussian filter for antialiasing is run with custom sigmas that are negligibly larger
+    than the scikit-image default at scaling factors <2, and smaller at scaling factors >2.
+    This should save some computation cost for large scaling steps. Cf. GitHub #3037.
+
+    Cannot resize along channel axis (would be nonsense).
+    Time is treated like space axes, so resize along t at your own risk.
+    """
+
+    class Interpolation(IntEnum):
+        NEAREST = 0
+        LINEAR = 1
+
+    semantics_to_interpolation = {
+        ImageTypes.Intensities: Interpolation.LINEAR,
+        ImageTypes.Labels: Interpolation.NEAREST,
+    }
+
+    required_min_padding = {  # minimum pixels of halo required for accurate interpolation
+        Interpolation.NEAREST: 0,
+        Interpolation.LINEAR: 2,
+    }
+
+    RawImage = InputSlot()
+    TargetShape = InputSlot()
+    InterpolationOrder = InputSlot(value=Interpolation.LINEAR)
+    ResizedImage = OutputSlot()
+
+    def __init__(
+        self,
+        parent=None,
+        graph=None,
+        RawImage: Slot = None,
+        TargetShape: Union[Tuple[int, ...], Slot, None] = None,
+        InterpolationOrder: Union[Interpolation, Slot, None] = None,
+        **kwargs,
+    ):
+        super().__init__(parent=parent, graph=graph, **kwargs)
+        self.scaling_factors: Union[NDArray, None] = None  # one factor per axis. Factor 2.0 = downscale to half size
+        self.antialiasing_sigmas: Union[NDArray, None] = None  # one sigma per axis
+        self.RawImage.setOrConnectIfAvailable(RawImage)
+        self.TargetShape.setOrConnectIfAvailable(TargetShape)
+        self.InterpolationOrder.setOrConnectIfAvailable(InterpolationOrder)
+
+    def setupOutputs(self):
+        if self.TargetShape.value == self.RawImage.meta.shape:
+            # Shortcut: no resizing needed
+            self.ResizedImage.connect(self.RawImage)
+        input_axiskeys = self.RawImage.meta.getAxisKeys()
+        assert len(input_axiskeys) == len(
+            self.TargetShape.value
+        ), f"Input image ({self.RawImage.meta.shape}) and target shape ({self.TargetShape.value}) must have same axes"
+        assert (
+            "c" not in input_axiskeys
+            or self.TargetShape.value[input_axiskeys.index("c")] == self.RawImage.meta.getTaggedShape()["c"]
+        ), "Cannot resize along channel axis"
+        if (
+            "t" in input_axiskeys
+            and self.TargetShape.value[input_axiskeys.index("t")] != self.RawImage.meta.getTaggedShape()["t"]
+        ):
+            logger.warning("Resizing along time axis. Are you sure this is what you want?")
```

Comment:
I'm not sure it's a good idea to allow for that "because we can". I at least hope that currently, for a user, it is not possible to reach that warning, right (because other components involved in export prevent that).

Reply:
I decided to allow it because it's not in principle wrong to scale along t, I'm just not anticipating that anyone wants to use ilastik to change the frame rate of their videos.

I'd leave that decision to if/when we decide to offer customisation of the exported scales...